zerver/context_processors.py

```diff
@@ -152,13 +152,29 @@ def login_context(request: HttpRequest) -> Dict[str, Any]:
         realm_description = get_realm_rendered_description(realm)
         realm_invite_required = realm.invite_required
 
+    def get_apple_locale() -> str:
+        supported_locales = ['ar_SA', 'ca_ES', 'cs_CZ', 'da_DK', 'de_DE',
+                             'en_US', 'es_ES', 'es_MX', 'fi_FI', 'fr_FR',
+                             'hr_HR', 'hu_HU', 'el_GR', 'nl_NL', 'no_NO',
+                             'it_IT', 'iw_IL', 'ja_JP', 'ko_KR', 'ms_MY',
+                             'pt_PT', 'ro_RO', 'ru_RU', 'sk_SK', 'sv_SE',
+                             'pl_PL', 'th_TH', 'tr_TR', 'uk_UA', 'vi_VI'
+                             'zh_CN', 'zh_HK', 'id_ID', 'pt_BR']
+
+        current_language_code = request.LANGUAGE_CODE
+        for locale in supported_locales:
+            if current_language_code in locale:
+                return locale
+        return 'en_US'
+
```

Comment:
This function would be great to have a few tests for.

To make that clean, one way is to pull it out as a top-level function, which can take `request.LANGUAGE_CODE` as an argument. Like this:
```py3
def get_apple_locale(django_language_code: str) -> str:
  # ...
```

There are also some changes I'm tempted to suggest to the algorithm. But that will be much easier to play with once the function is extracted and has some tests.

Reply:
Added one test which contains list of possible language codes and its corresponding locale, then asserts locale equal to the  return value  `get_apple_locale(language_code)`